### ISSUE TITLE
 Input 0 expected to have rank 4 but has rank 5

### ISSUE THREAD

**Issue by stevin-dong** [microsoft/onnxruntime#18313]
### Describe the issue

Conversion to Onnx model was successful, but there was an error during the initialization of onnxruntime：
onnxruntime.capi.onnxruntime_pybind11_state.Fail: [ONNXRuntimeError] : 1 : FAIL : Load model from dinet.onnx failed:Node (GridSample_178) Op (GridSample) [ShapeInferenceError] Input 0 expected to have rank 4 but has rank 5

### Target platform

ubuntu20.04

### Build script

provider_options_clone = [{'device_id': '0', 'trt_engine_cache_enable': 'True', 'trt_engine_cache_path': 'trt_engine_clone'}]
onnx_session_clone = ort.InferenceSession(f"{args.clone_onnx_path}", providers=['TensorrtExecutionProvider'], provider_options=provider_options_clone)

### Error / output

onnxruntime.capi.onnxruntime_pybind11_state.Fail: [ONNXRuntimeError] : 1 : FAIL : Load model from dinet.onnx failed:Node (GridSample_178) Op (GridSample) [ShapeInferenceError] Input 0 expected to have rank 4 but has rank 5

**Comment by skottmckay** [comment]
Can you share the model? Based on the error message it seems like there's a mismatch between the expected input for the GridSample node (4D) and what is being provided (5D). 

e.g. random example to illustrate the point, if I had a Reshape node that produced 5D output, and that output was provided to a GridSample node that only supports 4D input, that's an eeor.

How was the model converted? I suspect this issue might have been discovered if ONNX shape inferencing had been run as part of the conversion process.

One side note is that the GridSample operator was added in ONNX opset 16 and only allowed 4D input. In the most recent opset (20) the spec was updated to allow ND input. 

https://github.com/onnx/onnx/blob/main/docs/Changelog.md#GridSample-16
https://github.com/onnx/onnx/blob/main/docs/Operators.md#GridSample

If the input to GridSample is actually intended to be 5D the ONNX opset for the model would need to be 20 for that to be valid. FWIW support for the latest GridSample operator hasn't been added to ORT yet.

**Comment by stevin-dong** [comment]
> Can you share the model? Based on the error message it seems like there's a mismatch between the expected input for the GridSample node (4D) and what is being provided (5D).
> 
> e.g. random example to illustrate the point, if I had a Reshape node that produced 5D output, and that output was provided to a GridSample node that only supports 4D input, that's an eeor.
> 
> How was the model converted? I suspect this issue might have been discovered if ONNX shape inferencing had been run as part of the conversion process.
> 
> One side note is that the GridSample operator was added in ONNX opset 16 and only allowed 4D input. In the most recent opset (20) the spec was updated to allow ND input.
> 
> https://github.com/onnx/onnx/blob/main/docs/Changelog.md#GridSample-16 https://github.com/onnx/onnx/blob/main/docs/Operators.md#GridSample
> 
> If the input to GridSample is actually intended to be 5D the ONNX opset for the model would need to be 20 for that to be valid. FWIW support for the latest GridSample operator hasn't been added to ORT yet.

Thank you for your response! I currently need 5D input. Does this mean that ONNX Runtime does not yet support 5D input for GridSample? Because when I set opset_version to 20, it says there is no such version, and I can only choose 16 or 17.

**Comment by skottmckay** [comment]
Correct. The implementation needs to be updated to support ND input that was added to the spec in opset 20. most likely early next year unless there's an external contribution with the changes.

**Comment by abhishekmaroon5** [comment]
any update on this issue, i am facing the below issue. Please update once changes are released.

"onnxruntime.capi.onnxruntime_pybind11_state.Fail: [ONNXRuntimeError] : 1 : FAIL : Load model from dinet_new2.onnx failed:Node (/adaAT/GridSample) Op (GridSample) [ShapeInferenceError] Input 0 expected to have rank 4 but has rank 5
(abc) root@2f2e3da0c826:/home/dinet_tensorfrt# "

**Comment by ZelboK** [comment]
@skottmckay This seems like a fun one to solve. May I please pick it up? I can also do the CUDA implementation.
